### PR TITLE
#2694 [BUG][Typescript Axios] Content-type not set

### LINKED ISSUE #2694: [BUG][Typescript Axios] Content-type not set
##### Description

Typescript axios generators does not handle correctly multipart requests. Content-Type header is never set.

##### openapi-generator version

4.0.0-beta3

##### OpenAPI declaration file content or url

```yaml
openapi: 3.0.0

paths:
  'upload':
    post:
      summary: Upload File
      description: ''
      operationId: uploadFile
      requestBody:
        content:
          multipart/form-data:
            schema:
              $ref: '#/components/schemas/logoForCreation'
            encoding:
              medicalCertificateFileName:
                contentType: image/png, image/jpeg

        required:
          true
        description: body
      responses:
        201:
          description: Created or updated club's logo
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/photo'
components:
  schemas:
    logoForCreation:
      type: object
      required:
        - logoFileName
      properties:
        logoFileName:
          type: string
          format: binary
    photo:
      required:
        - location
      properties:
        location:
          type: string
```

##### Command line used for generation

openapi-generator generate -i swagger.yaml -g typescript-axios -o src/generated --skip-validate-spec

##### Steps to reproduce

Run the above command with the included yaml
Notice that the following expected line is missing:
localVarHeaderParameter['Content-Type'] = 'multipart/form-data';

##### Suggest a fix

multipartFormData mustache is used without vendorExtensions so it's not detected

### PR DESCRIPTION
**Pull request by julaudo**
fixes #2694 
[BUG][Typescript Axios] Content-type not set

### CODE REVIEW

**Comment by wing328** [comment]
@julaudo thanks for the PR. For the allOf issue, do you have a spec to help reproduce the issue?

**Comment by julaudo** [comment]
The allOf bug is another subject and should not be present is this pull request. To answer the question, the commit contains 2 specs to reproduce the bug (used by unit tests)

**Comment by macjohnny** [comment]
cc @nicokoenig

File: modules/openapi-generator/src/main/resources/typescript-axios/apiInner.mustache [hunk 1]
```diff
@@ -162,10 +162,10 @@ export const {{classname}}AxiosParamCreator = function (configuration?: Configur
             }
             {{/isListContainer}}
     {{/formParams}}{{/vendorExtensions}}
-    {{#hasFormParams}}{{#multipartFormData}}
+    {{#vendorExtensions}}{{#hasFormParams}}{{^multipartFormData}}
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/resources/typescript-axios/apiInner.mustache]: how does this solve issue https://github.com/OpenAPITools/openapi-generator/pull/2695?

**Comment by macjohnny** [comment]
any update here?

**Comment by macjohnny** [comment]
@julaudo can you please generate the samples?

**Comment by wing328** [comment]
@julaudo thanks for the PR, which has been included in the 4.1.0 release: https://twitter.com/oas_generator/status/1160000504455319553

**Pull request merged**